linux_os/guide/system/permissions/restrictions/poisoning/grub2_slub_debug_argument/policy/stig/shared.yml

```diff
@@ -1,39 +1,38 @@
-    {{{ full_name }}} must clear SLUB/SLAB objects to prevent use-after-free attacks.
+    {{{ full_name }}} must clear memory when it is freed to prevent use-after-free attacks.
 
 vuldiscussion: |-
     Some adversaries launch attacks with the intent of executing code in nonexecutable regions of memory or in memory locations that are prohibited. Security safeguards employed to protect memory include, for example, data execution prevention and address space layout randomization. Data execution prevention safeguards can be either hardware-enforced or software-enforced with hardware providing the greater strength of mechanism.
 
     Poisoning writes an arbitrary value to freed pages, so any modification or reference to that page after being freed or before being initialized will be detected and prevented. This prevents many types of use-after-free vulnerabilities at little performance cost. Also prevents leak of data and detection of corrupted memory.
 
-    SLAB objects are blocks of physically contiguous memory. SLUB is the unqueued SLAB allocator.
-
-
-
-checktext: |-
-    Verify that GRUB 2 is configured to enable poisoning of SLUB/SLAB objects to mitigate use-after-free vulnerabilities with the following commands:
+    init_on_free is a Linux kernel boot parameter that enhances security by initializing memory regions when they are freed, preventing data leakage. This process ensures that stale data in freed memory cannot be accessed by malicious programs.
 
-    Check that the current GRUB 2 configuration has poisoning of SLUB/SLAB objects enabled:
+    SLUB canaries add a randomized value (canary) at the end of SLUB-allocated objects to detect memory corruption caused by buffer overflows or underflows. Redzoning adds padding (red zones) around SLUB-allocated objects to detect overflows or underflows by triggering a fault when adjacent memory is accessed. SLUB canaries are often more efficient and provide stronger detection against buffer overflows compared to redzoning. SLUB canaries are supported in hardened Linux kernels like the ones provided by Linux-hardened.
 
-    $ sudo grubby --info=ALL | grep args | grep -v 'slub_debug=P'
+    SLAB objects are blocks of physically contiguous memory. SLUB is the unqueued SLAB allocator.
 
-    If any output is returned, this is a finding.
 
-    Check that poisoning of SLUB/SLAB objects is enabled by default to persist in kernel updates:
 
-    $ sudo grep slub_debug /etc/default/grub
+checktext: |-
+    Verify that GRUB2 is configured to mitigate use-after-free vulnerabilities by employing memory poisoning.
 
-    GRUB_CMDLINE_LINUX="slub_debug=P"
+    Inspect the "GRUB_CMDLINE_LINUX" entry of /etc/default/grub as follows:
+    $ sudo grep -i grub_cmdline_linux /etc/default/grub
+    GRUB_CMDLINE_LINUX="... init_on_free=1"
 
-    If "slub_debug" does not contain "P", is missing, or is commented out, this is a finding.
+    If "init_on_free=1" is missing or commented out, this is a finding.
```

Comment:
This change seems to warrant a new rule.

Reply:
Done.